core/metrics/src/main/scala/org/typelevel/otel4s/metrics/ObservableCounter.scala

```diff
@@ -16,4 +16,4 @@
 
 package org.typelevel.otel4s.metrics
 
-trait ObservableCounter[F[_], A]
+trait ObservableCounter
```

Comment:
Not sure what to do with these. It seems strange that they have these type parameters, when we really only use them in the callback. Now I tried to remove them, and that seems to aid in testing at least.